src/components/CreateShareStructure/ListShareClass.vue

```diff
@@ -1,14 +1,368 @@
+                      >
+                        <v-icon>mdi-menu-down</v-icon>
+                      </v-btn>
+                    </template>
+                    <v-list class="more-actions">
+                      <v-list-item
+                        class="actions-dropdown_item"
+                        :class="{ 'item-disabled': isMoveDisabled(row.index, 'up', index) }"
+                        @click="moveIndex(row.index, 'up', index)"
+                        :disabled="isMoveDisabled(row.index, 'up', index)"
+                      >
+                        <v-list-item-subtitle><v-icon>mdi-arrow-up</v-icon> Move Up</v-list-item-subtitle>
+                      </v-list-item>
+                      <v-list-item
+                        class="actions-dropdown_item"
+                        :class="{ 'item-disabled': isMoveDisabled(row.index, 'down', index) }"
+                        @click="moveIndex(row.index, 'down', index)"
+                        :disabled="isMoveDisabled(row.index, 'down', index)"
+                      >
+                        <v-list-item-subtitle><v-icon>mdi-arrow-down</v-icon> Move Down</v-list-item-subtitle>
+                      </v-list-item>
+                      <v-list-item class="actions-dropdown_item" @click="emitRemoveSeries(row.index, index)">
+                        <v-list-item-subtitle><v-icon>mdi-delete</v-icon> Remove</v-list-item-subtitle>
+                      </v-list-item>
+                    </v-list>
+                  </v-menu>
+                </span>
+            </div>
+          </td>
+        </tr>
+      </template>
+    </v-data-table>
+  </v-card>
 </template>
 
 <script lang="ts">
 // Libraries
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Prop, Vue, Emit } from 'vue-property-decorator'
+import 'array.prototype.move'
+
+// Components
+
+// Dialogs
+
+// Mixins
+
+// Interfaces
+
+    @Component({})
+export default class ListShareClass extends Vue {
+  @Prop({ default: () => [] })
+  private shareClasses: any
+
+  @Prop()
+  private componentDisabled: boolean
+
+  @Prop()
+  private isSummary: boolean
+
+  @Prop({ default: false })
+  private showErrorSummary: boolean
+
+  private headers: Array<any> = [
+    {
+      text: 'Name of Share Class or Series',
+      align: 'start',
+      sortable: false,
+      value: 'name'
+    },
+    { text: 'Maximum Number of Shares', value: 'maxNumberOfShares' },
+    { text: 'Par Value', value: 'parValue' },
+    { text: 'Currency', value: 'currency' },
+    { text: 'Special Rights or Restrictions', value: 'hasRightsOrRestrictions' },
+    { text: '', value: 'actions' }
+  ]
+
+  /**
+   * Adjust the priority of the list share class
+   * @param indexFrom The index of the class
+   * @param direction The direction of the move
+   * @param seriesIndex The index of the series
+   */
+  private moveIndex (indexFrom: number, direction: string, seriesIndex: number = -1): void {
```

Comment:
This is super-cool. Kudos for a nice, clean implementation!!!

Reply:
I didn't see a test for moving classes/series up or down. If there isn't one could you please add one? (Ideally the test would click the buttons instead of calling this method.)

Reply:
@cameron-freshworks  added a test for this

Reply:
Thanks @cameron-freshworks ! Looks good.